### PR TITLE
[Feature] #4415 Activities Overview for MMIS

### LINKED ISSUE #4415: [Feature] Update Activity Overview for MMIS
### Description and related issues
MMIS Activities don't have different funding types, so the funding type needs to be removed. Schedule has already been moved as part of #4371. The rich text fields are also different.

**Shared fields**
* Activity Name

**Fields for MMIS only**
* Activity Snapshot
* Problem Statement
* Proposed Solution

**Fields for HITECH only**
* Summary
* Description
* Alternatives
* Standards and Conditions

This ticket will require creating new set methods for the new MMIS fields actions/editActivity/overview.js.

### Mockups or link to Figma
https://www.figma.com/file/hJpKHKU6fz5J0Z7fisSwa2/eAPD-MMIS-2022?node-id=1388:12346&t=gKKGaYKaKXLCJ9Tk-0

### Acceptance criteria

| Given | When | Then |
| ----- | ---- | ---- |
|       |      |      |

### This task is done when…

- [ ] all acceptance criteria are met
- [ ] the Activity Overview component has been updated to handle HITECH and MMIS APD activities
- [ ] the new MMIS values are saving in mongo
- [ ] the new MMIS values are being displayed when the APD is opened again

### PR DESCRIPTION
**Pull request by amyd11**
Resolves #4415 

#### Direct MMIS changes
- Renamed the component `Overview.js` to `HITECHOverview.js` since this is specific to HITECH types
- Added an `MMISOverview` component since there is no overlap in the fields displayed (aside from the Activity Name now handled by the `NameForm` component) between `HITECHOverview` and `MMISOverview`
    - `MMISOverview` displays these sections:
        - Activity Name
        - Activity Snapshot
        - Problem Statement
        - Proposed Solution
- Created “new” `Overview.js` to be a wrapper component
    - Houses a mapping of the APD type to the chosen overview component (`MMISOverview` or `HITECHOverview`). This will allow us to add more types in the future
    - Sets the default type to be HITECH in case existing data does not have the `apdType` field
- Added redux actions for the new MMIS sections (`activitySnapshot`, `problemStatement`, `proposedSolution`)

#### Other indirect changes
- Split `NameAndFundingSourceForm` component into two: `NameForm` and `FundingSourceForm`
    - `NameForm` is used by `MMISOverview`
    - Both `NameForm` and `FundingSourceForm` are used by `HITECHOverview`
    - Renamed `NameAndFundingSourceForm.test.js` to `NameForm.test.js` since the test file was specific to testing the name (funding tests will be needed)
- Removed the NameAndFundingSource directory
    - `NameForm` and `FundingSourceForm` will live in the overview directory
    - Removed the only other component in this directory since it was unused (`NameAndFundingSourceReview` component and associated test file)

#### Content/Yaml changes
- Created `comprehensiveOverview.yaml` for content in the Comprehensive Overview section of an MMIS type
- Added content to the existing `overview.yaml` for an MMIS type

#### Dev Notes
- Further changes can be made to decouple name and funding source in the `nameAndFundingSourceSchema` schema

### Automated test cases written

| Given | When | Then | Type (jest, tap, cypress) |
| ----- | ---- | ---- | ------------------------- |
| Activity Overview for MMIS APD Type | Viewing existing data | Data displays in correct form field | jest |
| Activity Overview for MMIS APD Type | Viewing existing data | Data displays in correct form field | cypress |
| Activity Overview for MMIS APD Type with Admin check on | Fields are blank | Validation errors appear for each field | cypress |
| Activity Overview for MMIS APD Type with Admin check on | Fields are filled in | Validation errors disappear for each field | cypress |

### Steps to manually verify this change

1. Using [the figma](https://www.figma.com/file/hJpKHKU6fz5J0Z7fisSwa2/eAPD-MMIS-2022?node-id=1388%3A12346&t=3IIABZjkZzkxAArE-0), visit the Activity Overview page of an Activity for an MMIS APD to verify the page content matches the designs.
2. Ensure all the fields allow input and save entered data.
3. Turn on admin check to verify that validation errors display if any of the fields are blank.

### This pull request is ready to code review when

- [x] Automated tests are updated (and all tests are passing)
- [x] New automated test cases are documented above
- [x] Pull request has been labeled, if applicable with feature, content, bug,
      tests, refactor
- [ ] Associated OpenAPI documentation has been updated
- [x] The experience passes a basic manual accessibility audit (keyboard nav,
      screenreader, text scaling) OR an exemption is documented

### This pull request is ready to test when

- [ ] Code has been reviewed by someone other than the original author

### This pull request is ready to review when the QA has

- [x] Verified the functionality related to the change
- [x] Verified that the change works with Narrator on Windows
- [ ] Verified that the change works with VoiceOver on Mac
- [x] Verified all updated pages with the WAVE tool
- [x] Verified tab and keyboard navigation functionality

### This pull request can be merged when

- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: e2e/cypress/support/commands.js [hunk 1]
```diff
@@ -653,6 +653,24 @@ Cypress.Commands.add('turnOnAdminCheck', () => {
   });
 });
```
Comment (tbolt) [inline e2e/cypress/support/commands.js:655]: great additions

File: web/src/pages/apd/activities/overview/HITECHOverview.js [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+    trigger,
+    clearErrors,
+    formState: { errors }
+  } = useForm({
+    defaultValues: {
+      alternatives: alternatives,
+      description: description,
+      summary: summary
+    },
+    resolver: joiResolver(hitechActivityOverviewNoSCSchema)
```
Comment (tbolt) [inline web/src/pages/apd/activities/overview/HITECHOverview.js:45]: Might be a good idea to have a comment here or at the import spelling out SC = Standards and Conditions. What do you think?

**Review by tbolt — APPROVED** [review]
Nice work

File: web/src/pages/apd/activities/overview/HITECHOverview.js (hunk 2 above)
Comment (amyd11) [reply web/src/pages/apd/activities/overview/HITECHOverview.js:45]: I am all for airing on the side of being verbose for better code readability- I'll add in a comment to clarify that!

Comment (amyd11) [reply web/src/pages/apd/activities/overview/HITECHOverview.js:45]: I ended up adding a comment right at the source- the schema

File: web/src/pages/apd/activities/overview/MMISOverview.js [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+
+  useEffect(() => {
+    if (adminCheck) {
+      trigger();
+    } else {
+      clearErrors();
+    }
+  }, [adminCheck]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const snapshotLabel = useMemo(
```
Comment (thetif) [inline web/src/pages/apd/activities/overview/MMISOverview.js:60]: see comment above, you might be over optimizing with these useMemos

**Review by thetif — APPROVED** [review]
Looks great! I had a small concern with useMemo being used when it's not necessary, but everything works as expected

File: web/src/pages/apd/activities/overview/HITECHOverview.js [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+
+  useEffect(() => {
+    if (adminCheck) {
+      trigger();
+    } else {
+      clearErrors();
+    }
+  }, [adminCheck]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const overviewLabel = useMemo(
```
[the remark this answers is not in the document]
Comment (amyd11) [reply web/src/pages/apd/activities/overview/HITECHOverview.js]: I initially used it to follow the pattern of the HITECH version of the page and didn't think too deep about it at the time. Haha, Ty asked about it this morning which gave me some time to dig in- and totally agree that it feels unnecessary and more like clutter than anything. I’m glad you mentioned it too, which obviously gives more reason to pull it out- I’m pushing up the change to remove the useMemo from both HITECHOverview and MMISOverview. That should help clean it up!

**Review by Sun-Mountain — APPROVED** [review]
🚀

**Comment by mirano-darren** [comment]
Looks good to me. well done on the Cypress tests as well!
| Given | When | Then | Result |
| ----- | ---- | ---- | ------------------------- |
| a MMIS APD | on the Activity Overview page  | new page matches Figma design | ✅ |
| a HITECH APD | on the Activity Overview page  | page is the same as before | ✅ |
| a HITECH/MMIS APD | fields are filled in, navigating off and then on the page  | values are saved | ✅ |
| Activity Overview for HITECH/MMIS APD with Admin check on | Fields are blank | Validation errors appear for each field | ✅ |
| Activity Overview for HITECH/MMIS APD with Admin check on | Fields are filled in | Validation errors disappear for each field | ✅ |

**Review by SGilliamA1M — APPROVED** [review]
Looks good, well done!

**Review by jeromeleecms — APPROVED** [review]
LGTM!

**Pull request merged**